src/main/java/com/google/firebase/remoteconfig/RemoteConfigUtil.java

```diff
@@ -27,66 +27,52 @@
 
 final class RemoteConfigUtil {
 
+  private static final String ZULU_DATE_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSS000000'Z'";
+  private static final String ZULU_DATE_NO_FRAC_SECS_PATTERN = "yyyy-MM-dd'T'HH:mm:ss";
+  private static final String UTC_DATE_PATTERN = "EEE, dd MMM yyyy HH:mm:ss z";
+  private static final TimeZone UTC_TIME_ZONE = TimeZone.getTimeZone("UTC");
+
   static boolean isValidVersionNumber(String versionNumber) {
     return !Strings.isNullOrEmpty(versionNumber) && versionNumber.matches("^\\d+$");
   }
 
   static long convertToMilliseconds(String dateString) throws ParseException {
-    if (isUTCDateString(dateString)) {
+    try {
+      return convertFromUtcZuluFormat(dateString);
+    } catch (ParseException e) {
       return convertFromUtcDateFormat(dateString);
     }
-    return convertFromUtcZuluFormat(dateString);
   }
 
   static String convertToUtcZuluFormat(long millis) {
     // sample output date string: 2020-12-08T15:49:51.887878Z
     checkArgument(millis >= 0, "Milliseconds duration must not be negative");
-    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS000000'Z'");
-    dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+    SimpleDateFormat dateFormat = new SimpleDateFormat(ZULU_DATE_PATTERN);
+    dateFormat.setTimeZone(UTC_TIME_ZONE);
     return dateFormat.format(new Date(millis));
   }
 
   static String convertToUtcDateFormat(long millis) {
     // sample output date string: Tue, 08 Dec 2020 15:49:51 GMT
     checkArgument(millis >= 0, "Milliseconds duration must not be negative");
-    SimpleDateFormat dateFormat = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss z");
-    dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+    SimpleDateFormat dateFormat = new SimpleDateFormat(UTC_DATE_PATTERN);
+    dateFormat.setTimeZone(UTC_TIME_ZONE);
     return dateFormat.format(new Date(millis));
   }
 
   static long convertFromUtcZuluFormat(String dateString) throws ParseException {
     // sample input date string: 2020-12-08T15:49:51.887878Z
     checkArgument(!Strings.isNullOrEmpty(dateString), "Date string must not be null or empty");
-    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
-    dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+    SimpleDateFormat dateFormat = new SimpleDateFormat(ZULU_DATE_NO_FRAC_SECS_PATTERN);
```

Comment:
The example input doesn't really match the pattern. You might want to remove the fractional seconds before parsing the timestamp.

Reply:
Moved the date string formatting to this util method and added comments to clarify the steps.